tests/mocha/serializer_test.js

```diff
@@ -0,0 +1,1754 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+// TODO: Move this into samples as part of the dev-tools package.
+/**
+ * Constructs a serializer test.
+ * @param {string} title The title of this testcase.
+ * @param {string} xml The XML to use for the round-trip test.
+ * @constructor
+ */
+function SerializerTestCase(title, xml) {
+  this.title = title;
+  this.xml = xml;
+}
+SerializerTestCase.prototype = new testHelpers.TestCase();
+
+/**
+ * The XML we want to ensure round-trips through the serializer.
+ */
+SerializerTestCase.prototype.xml = '';
+
+
+/**
+ * Constructs a serializer test suite.
+ * @param {string} title The title of this test suite.
+ */
+function SerializerTestSuite(title) {
+  this.title = title;
+}
+SerializerTestSuite.prototype = new testHelpers.TestSuite();
+
+var Serializer = new SerializerTestSuite('Serializer');
+
+Serializer.Empty = new SerializerTestCase('Empty',
+    '<xml xmlns="https://developers.google.com/blockly/xml"></xml>'
+);
+Serializer.Data = new SerializerTestCase('Data',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="logic_negate" id="id******************" editable="false" x="42" y="42">' +
+    '<data>test data</data>' +
+    '</block>' +
+    '</xml>');
```

Comment:
This seems a little weird that these test cases (and the test cases defined later) are both on the TestSuite object directly and in the test cases. I would think it would only be necessary to add on `testCases` and i would just use a local variable when you are creating the test case.

Reply:
Yeah it is a bit weird haha. But the intention was to make it easy for external developers to skip/only suites/tests. If you only include things in the `testSuites` or `testCases` arrays then code for skipping becomes:
```
Serializer.testSuites[1].testSuites[0].testCases[4].skip = true;
```
as opposed to
```
Serializer.Fields.Angle.SmallestNumber.skip = true;
```

Reply:
Good idea! Because I'm not sure how this works with exporting either, and that'll be easier to figure out once it's actually in samples.